api/internalutils/stream/stream.go

```diff
@@ -372,3 +372,97 @@ func RateLimit[T any](stream Stream[T], wait func() error) Stream[T] {
 		wait:  wait,
 	}
 }
+
+// compareStreams is an adapter for merging two streams based on a `compare` function, it will get the next items from both streams and yield the result of streamA if the compare function returns true,
```

Comment:
```suggestion
// compareStreams is an adapter for merging two streams based on a compare function, it will get the next items from both streams and yield the result of streamA if the compare function returns true,
```

See https://tip.golang.org/doc/comment for an overview of doc comments - note that it doesn't recommend using backticks like this.